packages/cli/src/credentials/credentials.controller.ee.ts

```diff
@@ -190,15 +190,19 @@ EECredentialsController.put(
 			sharees_removed: amountRemoved,
 		});
 
+		const isEmailingEnabled = config.getEnv('userManagement.emails.mode') === 'smtp';
+
+		if (!isEmailingEnabled) return;
+
 		const recipients = await Container.get(UserRepository).getEmailsByIds(newShareeIds);
 
 		if (recipients.length === 0) return;
 
 		try {
 			await Container.get(UserManagementMailer).notifyCredentialsShared({
-				sharerFirstName: req.user.firstName,
+				sharer: { id: req.user.id, firstName: req.user.firstName },
+				newShareeIds,
 				credentialsName: credential.name,
-				recipientEmails: recipients.map(({ email }) => email),
 				baseUrl: Container.get(UrlService).getInstanceBaseUrl(),
 			});
 		} catch (error) {
```

Comment:
if we move this try-catch in `notifyCredentialsShared`, we can move the `InternalHooks.onEmailFailed` call in there as well. It just feels weird to move the success event in there, but leave the failure event out here.